stdlib/src/os/fstat.mojo

```diff
@@ -150,30 +150,43 @@ struct stat_result(Stringable):
         self.st_rdev = st_rdev
         self.st_flags = st_flags
 
+    @no_inline
+    fn write_to[W: Writer](self, inout writer: W):
+        """
+        Formats this path to the provided Writer.
+
+        Parameters:
+            W: A type conforming to the Writable trait.
+
+        Args:
+            writer: The object to write to.
+        """
+        writer.write("os.stat_result(")
+        writer.write("st_mode=", self.st_mode)
+        writer.write(", st_ino=", self.st_ino)
+        writer.write(", st_dev=", self.st_dev)
+        writer.write(", st_nlink=", self.st_nlink)
+        writer.write(", st_uid=", self.st_uid)
+        writer.write(", st_gid=", self.st_gid)
+        writer.write(", st_size=", self.st_size)
+        writer.write(", st_atime=", str(self.st_atimespec))
+        writer.write(", st_mtime=", str(self.st_mtimespec))
+        writer.write(", st_ctime=", str(self.st_ctimespec))
+        writer.write(", st_birthtime=", str(self.st_birthtimespec))
+        writer.write(", st_blocks=", self.st_blocks)
+        writer.write(", st_blksize=", self.st_blksize)
+        writer.write(", st_rdev=", self.st_rdev)
+        writer.write(", st_flags=", self.st_flags)
+        writer.write(")")
```

Comment:
Suggestion: 
The `String.write_bytes()` reserves the amount of bytes to join the next item every time, in the future we should be able to constrain traits and know if they implement for example `as_bytes()` and be able to reserve the total amount of bytes beforehand. So this suggestion doesn't make a difference currently but maybe in the future: 

these can all go in the same function call separated by commas. They will be formatted similarly by the Mojo formatter so it won't make much difference in the aesthetics other than the name of the field won't be directly to the left, still it's just a matter of personal preference and not performance currently :man_shrugging: .